packages/core/src/components/DataTable/index.tsx

```diff
@@ -0,0 +1,421 @@
+    this.props.selectOnRowClick && this.handleSelection(rowData)();
+  };
+
+  renderTableHeader(parentWidth: number) {
+    const {
+      editable,
+      instantEdit,
+      extraHeaderButtons,
+      rowHeight,
+      tableHeaderLabel,
+      tableHeaderHeight,
+    } = this.props;
+
+    const { editMode, selectedRows } = this.state;
+
+    return (
+      <TableHeader
+        editable={editable}
+        editMode={editMode}
+        instantEdit={instantEdit}
+        onEnableEditMode={this.enableEditMode}
+        onDisableEditMode={this.disableEditMode}
+        onCancelEditMode={this.cancelEditMode}
+        onEnactEdits={this.enactEdits}
+        extraHeaderButtons={extraHeaderButtons}
+        height={getHeight(rowHeight, tableHeaderHeight)}
+        selectedRows={selectedRows}
+        tableHeaderLabel={tableHeaderLabel}
+        width={this.props.width ? Math.min(this.props.width, parentWidth) : parentWidth}
+      />
+    );
+  }
+
+  rowGetter = (expandedDataList: ExpandedRow[]) => ({ index }: { index: number }) =>
+    expandedDataList[index];
+
+  render() {
+    const { columnHeaderHeight, expandable, rowHeight, selectable, styles } = this.props;
+
+    const {
+      sortedDataList,
+      expandedRows,
+      sortBy,
+      sortDirection,
+      editMode,
+      selectedRows,
+    } = this.state;
+
+    const expandedDataList = expandDataList(
+      sortedDataList,
+      expandedRows,
+      sortBy,
+      sortDirection,
+      this.keys,
+    );
+
+    return (
+      <div>
+        <AutoSizer disableHeight>
+          {({ width }: { width: number }) => this.renderTableHeader(width)}
+        </AutoSizer>
+        <div {...css(styles.table_container)}>
+          <AutoSizer disableHeight>
+            {({ width }: { width: number }) => (
+              <Table
+                height={this.props.height || 0}
+                width={this.props.width || width}
+                headerHeight={getHeight(rowHeight, columnHeaderHeight)}
+                rowCount={expandedDataList.length}
+                rowHeight={HEIGHT_TO_PX[rowHeight || "regular"]}
+                rowGetter={this.rowGetter(expandedDataList)}
+                rowStyle={this.rowStyles(expandedDataList)}
+                sort={this.sort}
+                sortBy={sortBy}
+                sortDirection={sortDirection}
+                headerRowRenderer={ColumnLabels(this.props)}
+                onRowClick={this.handleRowClick}
+              >
+                {expandable && renderExpandableColumn(styles, expandedRows, this.expandRow)}
+                {selectable && renderSelectableColumn(selectedRows, this.handleSelection, expandable)}
+                {renderDataColumns(this.keys, editMode, this.onEdit, this.props)}
+              </Table>
+            )}
+          </AutoSizer>
+        </div>
+      </div>
+    );
+  }
+}
+
+export default withStyles((theme: WithStylesProps['theme']) => ({
+    table_container: {
+      overflowX: 'scroll',
```

Comment:
can we set this to `auto`? `scroll` results in _always_ showing a scrollbar even when there is no overflow.

![image](https://user-images.githubusercontent.com/4496521/56932453-c2f63780-6a98-11e9-9e29-4ccf2969c318.png)